src/rpcserver/rpcserver.c

```diff
@@ -1115,11 +1187,13 @@ int main(int argc, const char *argv[])
 
     CHECK(0 == listen(server_fd, MAX_CONNECTIONS));
 
-    pthread_t thread;
 #ifdef __APPLE__
-    CHECK(0 == pthread_create(&thread, NULL, (void *(*)(void *))CFRunLoopRun, NULL));
+    pthread_t runloop_thread;
+    CHECK(0 == pthread_create(&runloop_thread, NULL, (void *(*)(void *))CFRunLoopRun, NULL));
 #endif // __APPLE__
 
+    signal(SIGCHLD, signal_handler);
```

Comment:
should be next to the `SIGPIPE` handler so the new process will inherit this too.
this signal handler is even more important since this is used to prevent the spawned zombied.

i also rather we'll ignore these signals entirely and remove the signal handler.